doc/user/content/overview/arangements.md

```diff
@@ -0,0 +1,149 @@
+
+When you create a materialized view and issue a query, Materialize creates a **dataflow**. A dataflow consists of the following:
+
+- An execution plan for the query
+- Independent computational components, or **operators**
+- A description of the connections between these components
+- Instructions on how to respond to data input, whether the input is static (fed in all at once) or dynamic (new data will continue to arrive after the initial dataflow execution)
+
+Once executed, the dataflow computes and stores the result of the SQL query in memory, polls the source for updates, and then incrementally updates the query results when new data arrives.
+
+The cost of maintaining a dataflow can be very different from the cost of executing a query once. Materialize queries are executed as long-lived stateful dataflows, and impose an ongoing cost of computation and memory. As input data changes, we want to respond quickly and without requiring a full query re-evaluation. This results in an optimization process that prioritizes the re-use of already materialized data.
+
+### Collections
+
+Materialize dataflows act on **collections** of data, [multisets](https://en.wikipedia.org/wiki/Multiset) that store each event in an update stream as a triple of `(data, time, diff)`.
+
+Term | Definition
+-----|-----------
+**data**  |  The record update.
+**time**  |  The logical timestamp of the update.
+**diff**  |  The type of update (`-1` for deletion, `1` for addition or upsert)
+
+Both inputs and outputs are expressed as collections.
+
+## Arrangements
+
+To transform an input into an output, operators
+build and maintain indexes on the input and output collections so that they can randomly access collection data in the future. Because queries can overlap, different operators often need to build the exact same indexes for multiple queries. Instead of asking operators to perform redundant work, Materialize simply builds the index once and maintains it in memory, sharing the required resources across all the operators that use the indexed data. The index is then effectively a sunk cost, and the cost of each query is determined only by the new work it introduces.
+
+We call the indexed update stream an **arrangement**.
+
+### Arrangement structure
+
+In the `(data, time, diff)` triples that make up the update stream, `data` is structured as a key-value pair.  Arrangements are indexed on the `data` keys.
+
+<!-- Is this correct? -->
+Note that this is not the same as the index created by the `CREATE INDEX` command. `CREATE INDEX` creates an index on a **source** or **table**. An arrangement is an index on the **update stream**. You cannot manually create an arrangement, although you can influence the shape of the arrangements created by Materialize by the way you structure queries and views.
```

Comment:
`CREATE INDEX` creates an index on a **source**, **view**, or **table**. 

The index created by the `CREATE INDEX` command is a Use 1 arrangement. 

Materialize will try to create as few Use 2 arrangements as possible by rendering the dataflow using Use 1 arrangements whenever possible.

In the join example below, if you create an index on `customer(c_custkey)`, Materialize will want to reuse the arrangement, so it will not apply the `c_mktsegment = 'BUILDING'` until after the join.

Reply:
Removed 

> Note that this is not the same as the index created by the `CREATE INDEX` command. `CREATE INDEX` creates an index on a **source** or **table**. An arrangement is an index on the **update stream**.